### ISSUE TITLE
[unimim] mismatched positional_embed about vit-large/14 for input resolution with 196

### ISSUE THREAD

**Issue by futureisatyourhand** [microsoft/unilm#921]
hello, for CLIP knowledge distilation paper, i.e.,A Unified View of Masked Image Modeling:
when the teacher is CLIP vit-large/14 for 196's input resolution, and the student is vit-base/16 for 224's input resolution, vit-large/14's  positional embed (i.e.,257) for CLIP  mismatch with the positional embed of our teacher (i.e., 197). How should I fix this to align with the paper.

Thanks very much!

**Comment by pengzhiliang** [comment]
Take it easy.

Please refer to [clip_model](https://github.com/microsoft/unilm/blob/master/beit2/vqkd_teacher/clip/model.py) in beit v2 codebase for details.
Specifically, the [interpolate_pos_encoding](https://github.com/microsoft/unilm/blob/a7a313dc0618e9f24a31af98d9183371314a03ba/beit2/vqkd_teacher/clip/model.py#L279) function is employed to resize the positional embedding.

**Comment by drkostas** [comment]
**tl;dr** I implemented MaskDistill from scratch in PyTorch, reproduced the paper's experiments, and open sourced the code and weights: https://github.com/drkostas/MaskDistill-PyTorch

I needed this for my research and the existing unofficial implementation (https://github.com/bwconrad/masked-distillation) was not giving me the expected results, so I ended up writing everything from scratch. ViT-Base pretrained for 300 epochs with CLIP teacher reproduces:

- Finetuning: 84.8% top-1 (paper reports 85.3%)
- Linear probe: 76.3% top-1
- Semantic segmentation: 52.6 mIoU (paper reports 53.8)
- k-NN: 75.6%
- Object detection: 44.4 bbox mAP
- Instance segmentation: 40.1 segm mAP

All checkpoints on HuggingFace: https://huggingface.co/drkostas/MaskDistill-ViT-Base
